mithril-aggregator/src/multi_signer.rs

```diff
@@ -43,31 +51,56 @@ impl MultiSignerImpl {
         debug!("New MultiSignerImpl created");
         Self { epoch_service }
     }
-}
 
-#[async_trait]
-impl MultiSigner for MultiSignerImpl {
-    /// Verify a single signature
-    async fn verify_single_signature(
+    fn run_verify_single_signature(
         &self,
-        message: &entities::ProtocolMessage,
+        message: &str,
         single_signature: &entities::SingleSignatures,
+        protocol_multi_signer: &ProtocolMultiSigner,
     ) -> StdResult<()> {
         debug!(
             "Verify single signature from {} at indexes {:?} for message {:?}",
             single_signature.party_id, single_signature.won_indexes, message
         );
 
+        protocol_multi_signer
+            .verify_single_signature(&message, single_signature)
+            .with_context(|| {
+                format!("Multi Signer can not verify single signature for message '{message:?}'")
+            })
+    }
+}
+
+#[async_trait]
+impl MultiSigner for MultiSignerImpl {
+    /// Verify a single signature
+    async fn verify_single_signature(
+        &self,
+        message: &str,
+        single_signature: &entities::SingleSignatures,
+    ) -> StdResult<()> {
         let epoch_service = self.epoch_service.read().await;
         let protocol_multi_signer = epoch_service.protocol_multi_signer().with_context(|| {
             "Multi Signer could not get protocol multi-signer from epoch service"
         })?;
 
-        protocol_multi_signer
-            .verify_single_signature(message, single_signature)
-            .with_context(|| {
-                format!("Multi Signer can not verify single signature for message '{message:?}'")
-            })
+        self.run_verify_single_signature(message, single_signature, protocol_multi_signer)
+    }
+
+    async fn verify_single_signature_for_next_epoch(
```

Comment:
Maybe using `next_protocol_multi_signer` instead of `epoch` in the function name would be more accurate?
```suggestion
    async fn verify_single_signature_for_next_protocol_multi_signer(
```